internal/pkg/cli/app_show.go

```diff
@@ -0,0 +1,312 @@
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/describe"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	applicationShowProjectNamePrompt     = "Which project's applications would you like to show?"
+	applicationShowProjectNameHelpPrompt = "A project groups all of your applications together."
+	applicationShowAppNamePrompt         = "Which application of %s would you like to show?"
+	applicationShowAppNameHelpPrompt     = "The detail of an application will be shown (e.g., endpoint URL, CPU, Memory)."
+)
+
+// ShowAppOpts contains the fields to collect for showing an application.
+type ShowAppOpts struct {
+	ShouldOutputJSON bool
+
+	proj *archer.Project
+	app  *archer.Application
+	envs []*archer.Environment
+	uris []*describe.WebAppURI
+
+	appName string
+
+	projectLister archer.ProjectLister
+	projectGetter archer.ProjectGetter
+	appLister     archer.ApplicationLister
+	appGetter     archer.ApplicationGetter
+	envLister     archer.EnvironmentLister
+	identifier    ResourceIdentifier
+
+	w io.Writer
+
+	*GlobalOpts
+}
+
+// BuildAppShowCmd builds the command for showing applications in a project.
+func BuildAppShowCmd() *cobra.Command {
+	opts := ShowAppOpts{
+		w:          os.Stdout,
+		GlobalOpts: NewGlobalOpts(),
+	}
+	cmd := &cobra.Command{
+		Use:   "show",
+		Short: "Shows all the applications in a project",
+		Example: `
+  Shows detail for the application my-app
+  /code $ ecs-preview app show -a my-app`,
+		PreRunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			ssmStore, err := store.New()
+			if err != nil {
+				return fmt.Errorf("connect to environment datastore: %w", err)
+			}
+			opts.projectLister = ssmStore
+			opts.projectGetter = ssmStore
+			opts.appLister = ssmStore
+			opts.appGetter = ssmStore
+			opts.envLister = ssmStore
+
+			return nil
+		}),
+		RunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			if err := opts.Ask(); err != nil {
+				return err
+			}
+			identifier, err := describe.NewWebAppDescriber(opts.ProjectName(), opts.appName)
+			if err != nil {
+				return fmt.Errorf("creating identifier for application %s in project %s: %w", opts.appName, opts.ProjectName(), err)
+			}
+			opts.identifier = identifier
+			if err := opts.Execute(); err != nil {
+				return err
+			}
+
+			return nil
+		}),
+	}
+	// The flags bound by viper are available to all sub-commands through viper.GetString({flagName})
+	cmd.Flags().StringVarP(&opts.appName, appFlag, appFlagShort, "", appFlagDescription)
+	cmd.Flags().BoolVar(&opts.ShouldOutputJSON, jsonFlag, false, jsonFlagDescription)
+	cmd.PersistentFlags().StringP(projectFlag, projectFlagShort, "" /* default */, projectFlagDescription)
+	viper.BindPFlag(projectFlag, cmd.PersistentFlags().Lookup(projectFlag))
```

Comment:
I don't think they need to be Persistent Flags unless a sub-command will be created (such as `app show other-command`), so we can just use `cmd.Flags()`